include/deal.II/fe/fe_update_flags.h

```diff
@@ -240,114 +302,143 @@ inline StreamType &
   return s;
 }
 
+namespace internal
+{
+  constexpr UpdateFlags
+  make_update_flags(int field)
+  {
+    return UpdateFlags(field);
+  }
+} // namespace internal
 
-/**
- * Global operator which returns an object in which all bits are set which are
- * either set in the first or the second argument. This operator exists since
- * if it did not then the result of the bit-or <tt>operator |</tt> would be an
- * integer which would in turn trigger a compiler warning when we tried to
- * assign it to an object of type UpdateFlags.
- *
- * @ref UpdateFlags
- */
-inline UpdateFlags
-operator|(const UpdateFlags f1, const UpdateFlags f2)
+inline constexpr UpdateFlags::UpdateFlags(int i)
+  : field(i)
+{}
+
+inline constexpr UpdateFlags
+UpdateFlags::operator|(const UpdateFlags in) const
 {
-  return static_cast<UpdateFlags>(static_cast<unsigned int>(f1) |
-                                  static_cast<unsigned int>(f2));
+  return internal::make_update_flags(field | in.field);
 }
 
 
 
-/**
- * Global operator which sets the bits from the second argument also in the
- * first one.
- *
- * @ref UpdateFlags
- */
-inline UpdateFlags &
-operator|=(UpdateFlags &f1, const UpdateFlags f2)
+inline constexpr UpdateFlags &
+UpdateFlags::operator|=(const UpdateFlags in)
 {
-  f1 = f1 | f2;
-  return f1;
+  field = field | in.field;
+  return *this;
 }
 
 
-/**
- * Global operator which returns an object in which all bits are set which are
- * set in the first as well as the second argument. This operator exists since
- * if it did not then the result of the bit-and <tt>operator &</tt> would be
- * an integer which would in turn trigger a compiler warning when we tried to
- * assign it to an object of type UpdateFlags.
- *
- * @ref UpdateFlags
- */
-inline UpdateFlags
-operator&(const UpdateFlags f1, const UpdateFlags f2)
+
+inline constexpr UpdateFlags
+UpdateFlags::operator&(const UpdateFlags in) const
 {
-  return static_cast<UpdateFlags>(static_cast<unsigned int>(f1) &
-                                  static_cast<unsigned int>(f2));
+  return internal::make_update_flags(field & in.field);
 }
 
 
-/**
- * Global operator which clears all the bits in the first argument if they are
- * not also set in the second argument.
- *
- * @ref UpdateFlags
- */
-inline UpdateFlags &
-operator&=(UpdateFlags &f1, const UpdateFlags f2)
+
+inline constexpr UpdateFlags &
+UpdateFlags::operator&=(const UpdateFlags in)
+{
+  field = field & in.field;
+  return *this;
+}
+
+
+
+inline constexpr bool
+UpdateFlags::operator==(const UpdateFlags in) const
 {
-  f1 = f1 & f2;
-  return f1;
+  return field == in.field;
 }
 
 
 
+inline constexpr bool
+UpdateFlags::operator!=(const UpdateFlags in) const
+{
+  return field != in.field;
+}
+
+
+inline constexpr bool
+UpdateFlags::contains(const UpdateFlags in) const
+{
+  return (field & in.field) != 0;
+}
+
+// Import all flags into the global namespace.
+constexpr UpdateFlags update_default;
+constexpr UpdateFlags update_values;
```

Comment:
Don't these have to be initialized somehow? I don't see where the "import" actually happens :-)